app/models/suite_importer.rb

```diff
@@ -0,0 +1,132 @@
+# frozen_string_literal: true
+#
+# Form Object for Suite Importing
+class SuiteImporter
+  include ActiveModel::Model
+  require 'csv'
+
+  HEADER = %w(number singles doubles).freeze
+
+  attr_reader :building, :file
+
+  # Initialize a new SuiteImporter and call #import on it
+  def self.import(**params)
+    new(**params).import
+  end
+
+  # Initialize a new SuiteImporter and call #import on it
+  #
+  # @param [String] file The path to the CSV
+  # @param [Building] building The building to put the suites in
+  def initialize(file: nil, building:)
+    @building = building
+    @file = file
+    @successes = []
+    @failures = []
+  end
+
+  # Read in a CSV of suite information and create suites and rooms accordingly
+  # The CSV should be formatted with a header Number, Single, Double, where
+  # Single is the number of single rooms in the suite and Double is the number
+  # of double rooms.
+  #
+  # @return [Hash{Symbol=>nil,Hash}] A hash with flash messages to be set.
+  def import
+    return error('No file uploaded') unless file
```

Comment:
so... ideally we would be using ActiveModel validations for this stuff, e.g. `validate :file, presence: true` and `validates :correct_header?` (probably renamed without the question mark). Then we'd just return `error(errors.full_messages.join(', ')) unless valid?` at the start of `#submit`.